### PR TITLE
Add recipe for bayes-filters-lib C++ library

### PR DESCRIPTION
**Pull request by traversaro**
[bayes-filters-lib](https://github.com/robotology/bayes-filters-lib) is C++ recursive Bayesian estimation library.

The package name contains the `lib` prefix as requested in https://github.com/conda-forge/staged-recipes/pull/19764#pullrequestreview-1085836936 for C/C++ libraries.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
@GiulioRomualdi @xEnVrE are you interested in being co-mantainers of this?

**Comment by GiulioRomualdi** [comment]
> @GiulioRomualdi @xEnVrE are you interested in being co-mantainers of this?


Green flag for me!

**Comment by xEnVrE** [comment]
Fine with me as well @traversaro

**Comment by traversaro** [comment]
Thanks @GiulioRomualdi , @xEnVrE . @conda-forge/help-c-cpp The PR is ready for review.

File: recipes/bayes-filters-lib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - pkg-config
+    - ninja
+  host:
+    - eigen
+
+test:
+  commands:
```
Comment (carterbox) [inline recipes/bayes-filters-lib/meta.yaml:28]: Please include check for cmake config files.
```suggestion
  commands:
    - test -f $PREFIX/lib/cmake/BayesFilters/BayesFiltersConfig.cmake  # [unix]
    - if not exist %PREFIX%\\Library\\CMake\\BayesFiltersConfig.cmake  # [win]
```

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/bayes-filters-lib/meta.yaml (hunk 1 above)
Comment (traversaro) [reply recipes/bayes-filters-lib/meta.yaml:28]: Done.

**Pull request merged**